### DISCUSSION TITLE
Issue with telegram

### DISCUSSION THREAD

**Discussion by orikuOlarotimi** [hlf20010508/telegram-onedrive#17] [Q&A] [answered]
I've submitted my telegram code to the url but still i get the warning that you haven't logged in to telegram and it continues to say logining to telegram and then you haven't logged in to telegram
Any solution

**Comment by hlf20010508** [comment]
What happened after you submit telegram code? Did OneDrive authorization url show?

**Comment by orikuOlarotimi** [comment]
No sir

**Comment by orikuOlarotimi** [comment]
It just kept on telling me you haven't logged on to telegram and later logging into telegram after and that's where it ends

**Comment by hlf20010508** [comment]
Are you sure you submitted the right code?
Recreate docker container and try again.

**Comment by hlf20010508** [comment]
> It just kept on telling me you haven't logged on to telegram and later logging into telegram after and that's where it ends

This happens when you've submitted a wrong code and didn't recreate docker container.
Make sure you submit the latest code. Don't submit old code by mistake.
And make sure you don't have Two-Step Verification. I've not tested it yet.

**Comment by orikuOlarotimi** [comment]
The problem is that the bot goes on a loop so once it tells me to click on the url immediately it sends me logging in to telegram and then it says you've not yet logged in to telegram and then does same thing over some time

**Comment by hlf20010508** [comment]
> The problem is that the bot goes on a loop so once it tells me to click on the url immediately it sends me logging in to telegram and then it says you've not yet logged in to telegram and then does same thing over some time

Have you recreate docker container?

**Comment by orikuOlarotimi** [comment]
I should delete it the existing container? And build a new one ?

**Comment by hlf20010508** [comment]
> I should delete it the existing container? And build a new one ?

yes

**Comment by orikuOlarotimi** [comment]
Ok

**Comment by orikuOlarotimi** [comment]
I'm coming

**Comment by orikuOlarotimi** [comment]
I'm currently using your docker image

**Comment by orikuOlarotimi** [comment]
Should i build my docker image?

**Comment by hlf20010508** [comment]
> Should i build my docker image?

no

**Comment by hlf20010508** [comment]
just make sure your phone number is correct, and use the latest code

**Comment by orikuOlarotimi** [comment]
Should i use the international format like +234

**Comment by hlf20010508** [comment]
yes, +234xxxxxxxxx

**Comment by orikuOlarotimi** [comment]
Ok
I'm going to start the docker container now

**Comment by hlf20010508** [comment]
any feedback?

**Comment by orikuOlarotimi** [comment]
I'm still having same result
I'm using docker compose up -d

**Comment by hlf20010508** [comment]
show me the log. use `docker log`

**Comment by orikuOlarotimi** [comment]
Ok

**Comment by orikuOlarotimi** [comment]
Unhandled exception on wrapper
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 203, in _new_conn
telegram-onedrive  |     sock = connection.create_connection(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
telegram-onedrive  |     raise err
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
telegram-onedrive  |     sock.connect(sa)
telegram-onedrive  | ConnectionRefusedError: [Errno 111] Connection refused
telegram-onedrive  |
telegram-onedrive  | The above exception was the direct cause of the following exception:
telegram-onedrive  |
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 791, in urlopen
telegram-onedrive  |     response = self._make_request(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 492, in _make_request
telegram-onedrive  |     raise new_e
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 468, in _make_request
telegram-onedrive  |     self._validate_conn(conn)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1097, in _validate_conn
telegram-onedrive  |     conn.connect()
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 611, in connect
telegram-onedrive  |     self.sock = sock = self._new_conn()
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 218, in _new_conn
telegram-onedrive  |     raise NewConnectionError(
telegram-onedrive  | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f2d50da6640>: Failed to establish a new connection: [Errno 111] Connection refused
telegram-onedrive  |
telegram-onedrive  | The above exception was the direct cause of the following exception:
telegram-onedrive  |
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 486, in send
telegram-onedrive  |     resp = conn.urlopen(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 845, in urlopen
telegram-onedrive  |     retries = retries.increment(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 515, in increment
telegram-onedrive  |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
telegram-onedrive  | urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /tg (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f2d50da6640>: Failed to establish a new connection: [Errno 111] Connection refused'))
telegram-onedrive  |
telegram-onedrive  | During handling of the above exception, another exception occurred:
telegram-onedrive  |
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/updates.py", line 570, in _dispatch_update
telegram-onedrive  |     await callback(event)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 101, in wrapper
telegram-onedrive  |     return await func(event, *args, **kwargs)
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 77, in auth_handler
telegram-onedrive  |     tg_client = await tg_client.start(tg_user_phone, code_callback=tg_code_callback)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/auth.py", line 189, in _start
telegram-onedrive  |     value = await value
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 63, in __call__
telegram-onedrive  |     return await self.code
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 58, in code
telegram-onedrive  |     return await self.tg_code
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 29, in tg_code
telegram-onedrive  |     res = requests.get(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 73, in get
telegram-onedrive  |     return request("get", url, params=params, **kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 59, in request
telegram-onedrive  |     return session.request(method=method, url=url, **kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 589, in request
telegram-onedrive  |     resp = self.send(prep, **send_kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 703, in send
telegram-onedrive  |     r = adapter.send(request, **kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 519, in send
telegram-onedrive  |     raise ConnectionError(e, request=request)
telegram-onedrive  | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /tg (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f2d50da6640>: Failed to establish a new connection: [Errno 111] Connection refused'))
telegram-onedrive  | Unhandled exception on wrapper
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/updates.py", line 570, in _dispatch_update
telegram-onedrive  |     await callback(event)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 101, in wrapper
telegram-onedrive  |     return await func(event, *args, **kwargs)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 111, in wrapper
telegram-onedrive  |     await res_not_login(event)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 119, in res_not_login
telegram-onedrive  |     await auth_handler(event, propagate=True)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 101, in wrapper
telegram-onedrive  |     return await func(event, *args, **kwargs)
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 77, in auth_handler
telegram-onedrive  |     tg_client = await tg_client.start(tg_user_phone, code_callback=tg_code_callback)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/auth.py", line 184, in _start
telegram-onedrive  |     await self.send_code_request(phone, force_sms=force_sms)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/auth.py", line 460, in send_code_request
telegram-onedrive  |     result = await self(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/users.py", line 30, in __call__
telegram-onedrive  |     return await self._call(self._sender, request, ordered=ordered)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/users.py", line 87, in _call
telegram-onedrive  |     result = await future
telegram-onedrive  | telethon.errors.rpcerrorlist.SendCodeUnavailableError: Returned when all available options for this type of number were already used (e.g. flash-call, then SMS, then this error might be returned to trigger a second resend) (caused by ResendCodeRequest)

**Comment by orikuOlarotimi** [comment]
Unhandled exception on wrapper
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 203, in _new_conn
telegram-onedrive  |     sock = connection.create_connection(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
telegram-onedrive  |     raise err
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
telegram-onedrive  |     sock.connect(sa)
telegram-onedrive  | ConnectionRefusedError: [Errno 111] Connection refused
telegram-onedrive  |
telegram-onedrive  | The above exception was the direct cause of the following exception:
telegram-onedrive  |
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 791, in urlopen
telegram-onedrive  |     response = self._make_request(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 492, in _make_request
telegram-onedrive  |     raise new_e
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 468, in _make_request
telegram-onedrive  |     self._validate_conn(conn)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1097, in _validate_conn
telegram-onedrive  |     conn.connect()
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 611, in connect
telegram-onedrive  |     self.sock = sock = self._new_conn()
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 218, in _new_conn
telegram-onedrive  |     raise NewConnectionError(
telegram-onedrive  | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f2d50da6640>: Failed to establish a new connection: [Errno 111] Connection refused
telegram-onedrive  |
telegram-onedrive  | The above exception was the direct cause of the following exception:
telegram-onedrive  |
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 486, in send
telegram-onedrive  |     resp = conn.urlopen(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 845, in urlopen
telegram-onedrive  |     retries = retries.increment(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 515, in increment
telegram-onedrive  |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
telegram-onedrive  | urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /tg (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f2d50da6640>: Failed to establish a new connection: [Errno 111] Connection refused'))
telegram-onedrive  |
telegram-onedrive  | During handling of the above exception, another exception occurred:
telegram-onedrive  |
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/updates.py", line 570, in _dispatch_update
telegram-onedrive  |     await callback(event)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 101, in wrapper
telegram-onedrive  |     return await func(event, *args, **kwargs)
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 77, in auth_handler
telegram-onedrive  |     tg_client = await tg_client.start(tg_user_phone, code_callback=tg_code_callback)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/auth.py", line 189, in _start
telegram-onedrive  |     value = await value
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 63, in __call__
telegram-onedrive  |     return await self.code
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 58, in code
telegram-onedrive  |     return await self.tg_code
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 29, in tg_code
telegram-onedrive  |     res = requests.get(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 73, in get
telegram-onedrive  |     return request("get", url, params=params, **kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 59, in request
telegram-onedrive  |     return session.request(method=method, url=url, **kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 589, in request
telegram-onedrive  |     resp = self.send(prep, **send_kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 703, in send
telegram-onedrive  |     r = adapter.send(request, **kwargs)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 519, in send
telegram-onedrive  |     raise ConnectionError(e, request=request)
telegram-onedrive  | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /tg (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f2d50da6640>: Failed to establish a new connection: [Errno 111] Connection refused'))
telegram-onedrive  | Unhandled exception on wrapper
telegram-onedrive  | Traceback (most recent call last):
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/updates.py", line 570, in _dispatch_update
telegram-onedrive  |     await callback(event)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 101, in wrapper
telegram-onedrive  |     return await func(event, *args, **kwargs)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 111, in wrapper
telegram-onedrive  |     await res_not_login(event)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 119, in res_not_login
telegram-onedrive  |     await auth_handler(event, propagate=True)
telegram-onedrive  |   File "/telegram-onedrive/modules/utils.py", line 101, in wrapper
telegram-onedrive  |     return await func(event, *args, **kwargs)
telegram-onedrive  |   File "/telegram-onedrive/modules/handlers/auth.py", line 77, in auth_handler
telegram-onedrive  |     tg_client = await tg_client.start(tg_user_phone, code_callback=tg_code_callback)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/auth.py", line 184, in _start
telegram-onedrive  |     await self.send_code_request(phone, force_sms=force_sms)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/auth.py", line 460, in send_code_request
telegram-onedrive  |     result = await self(
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/users.py", line 30, in __call__
telegram-onedrive  |     return await self._call(self._sender, request, ordered=ordered)
telegram-onedrive  |   File "/usr/local/lib/python3.8/site-packages/telethon/client/users.py", line 87, in _call
telegram-onedrive  |     result = await future
telegram-onedrive  | telethon.errors.rpcerrorlist.SendCodeUnavailableError: Returned when all available options for this type of number were already used (e.g. flash-call, then SMS, then this error might be returned to trigger a second resend) (caused by ResendCodeRequest)
telegram-onedrive  |  * Serving Flask app 'auth_server'
telegram-onedrive  |  * Debug mode: off
telegram-onedrive  |  * Serving Flask app 'auth_server'
telegram-onedrive  |  * Debug mode: off
telegram-onedrive  |  * Serving Flask app 'auth_server'
telegram-onedrive  |  * Debug mode: off
telegram-onedrive  | WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
telegram-onedrive  |  * Running on all addresses (0.0.0.0)
telegram-onedrive  |  * Running on https://127.0.0.1:8080
telegram-onedrive  |  * Running on https://192.168.65.3:8080
telegram-onedrive  | Press CTRL+C to quit
telegram-onedrive  | WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
telegram-onedrive  |  * Running on all addresses (0.0.0.0)
telegram-onedrive  |  * Running on https://127.0.0.1:8080
telegram-onedrive  |  * Running on https://192.168.65.3:8080
telegram-onedrive  | Press CTRL+C to quit
telegram-onedrive  | WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
telegram-onedrive  |  * Running on all addresses (0.0.0.0)
telegram-onedrive  |  * Running on https://127.0.0.1:8080
telegram-onedrive  |  * Running on https://192.168.65.3:8080
telegram-onedrive  | Press CTRL+C to quit

**Comment by hlf20010508** [comment]
Are you running on a server or a PC?

**Comment by orikuOlarotimi** [comment]
on 
a pc

**Comment by hlf20010508** [comment]
Ok, try using
```
ports:
  - 8080:8080
```

instead of 
```
network_mode: host
```

in your docker-compose.yml

**Comment by hlf20010508** [comment]
I did found this problem in [https://github.com/docker/compose/issues/3800](https://github.com/docker/compose/issues/3800). But I only have macOS, so I can't test it on Windows.

**Comment by orikuOlarotimi** [comment]
okk ill try these later

**Comment by hlf20010508** [comment]
> Ok, try using
> 
> ```
> ports:
>   - 8080:8080
> ```
> 
> instead of
> 
> ```
> network_mode: host
> ```
> 
> in your docker-compose.yaml

Good news. I've installed Docker Desktop 4.4.4 on Windows 10 LTSC 21H2 and tested it, and it worked.

**Comment by orikuOlarotimi** [comment]
I'll try it now

**Comment by orikuOlarotimi** [comment]
This site can't be reached 
Localhost unexpectedly closed the connection

**Comment by orikuOlarotimi** [comment]
Ok it worked but I'm now getting an error with one drive

**Comment by orikuOlarotimi** [comment]
We're unable to complete your request
invalid_request: The provided value for the input parameter 'redirect_uri' is not valid. The expected value is a URI which matches a redirect URI registered for this client application.

**Comment by hlf20010508** [comment]
> We're unable to complete your request
> 
> invalid_request: The provided value for the input parameter 'redirect_uri' is not valid. The expected value is a URI which matches a redirect URI registered for this client application.

What's your Redirect URI on Azure Portal?

And what's your server_uri in your docker-compose.yml?

**Comment by orikuOlarotimi** [comment]
Https://localhost:8080

**Comment by hlf20010508** [comment]
Set `server_uri` to `https://127.0.0.1:8080`.
Set Redirect URI on Azure Portal to `https://127.0.0.1:8080/auth`

**Comment by orikuOlarotimi** [comment]
https://127.0.0.1:8080 doesn't work with my oc but localhost does and I don't know why

**Comment by orikuOlarotimi** [comment]
That's why i use localhost

**Comment by hlf20010508** [comment]
your Redirect URI on Azure Portal is `https://localhost:8080/auth`?

**Comment by orikuOlarotimi** [comment]
I'll update it now

**Comment by orikuOlarotimi** [comment, marked as answer]
I didn't add the /auth

**Comment by hlf20010508** [comment]
Any feedback?
